server/cluster/cluster.go

```diff
@@ -665,6 +664,55 @@ func (c *RaftCluster) processRegionHeartbeat(region *core.RegionInfo) error {
 		time.Sleep(500 * time.Millisecond)
 	})
 
+	return c.saveRegion(isNew, saveKV, saveCache, needSync, region, origin)
+}
+
+// processRegionHeartbeatResponse updates the region information thought region split.
+// It will update peer that not exist in the cache, and add new peers that join in.
+func (c *RaftCluster) processRegionSpiltReport(regions []*metapb.Region) (err error) {
+	if err = checkSplitRegions(regions); err != nil {
+		return err
+	}
+	c.RLock()
+	coreCluster := c.core
+	c.RUnlock()
+	total := len(regions) - 1
+	for i := range regions {
+		// It should update rightmost because it will override by left.
+		v := regions[total-i]
+		if v.Peers == nil {
+			err = errors.Wrap(err, "region peer is nil")
```

Comment:
Will it wrap multiple times?

Reply:
yes, the error message will add region-id.

Reply:
I think it may have something unexpected with the current implementation.